lib/analytics/ga4/google_analytics_service.rb

```diff
@@ -0,0 +1,59 @@
+require "google/analytics/data/v1beta"
+require_relative "./request"
+require_relative "./page_data"
+require_relative "./client"
+
+require "dotenv"
+
+Dotenv.load
+
+module Analytics
+  module Ga4
+    class GoogleAnalyticsService
+      attr_accessor :ga_client, :all_data
+
+      OFFSET = 0
+      LIMIT = 10_000
+      OFFSET_INCREMENT = 10_000
+
+      def initialize
+        @ga_client = Analytics::Ga4::Client.new
+        @all_data = []
+      end
+
+      def get_paginated_data
+        # https://developers.google.com/analytics/devguides/reporting/data/v1/basics#navigate_long_reports
+        offset = OFFSET
+
+        loop do
+          data = get_data(offset)
+          break if data.nil? || data[:rows].nil? || data[:rows].empty?
+
+          all_data << format_all_data(data[:rows])
+
+          offset += OFFSET_INCREMENT
+        end
+
+        all_data.flatten!
+      end
+
+    private
+
+      def format_all_data(data)
+        data.map do |row|
+          path = row[:dimension_values].first[:value]
+          title = row[:dimension_values].last[:value]
+          page_views = row[:metric_values].first[:value]
+
+          Analytics::Ga4::PageData.new(path, title, page_views)
```

Comment:
```suggestion
          PageData.new(path, title, page_views)
```

again already in this namespace.